lib/core.ps1

```diff
@@ -635,12 +635,12 @@ function shim($path, $global, $name, $arg) {
         @(
             "#!/bin/sh",
             "# $resolved_path",
-            "if command -v pwsh.exe &> /dev/null; then",
-            "    pwsh -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"",
+            "if command -v pwsh.exe > /dev/null 2>&1; then",
+            "    pwsh.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"",
             "else",
-            "    powershell -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"",
+            "    powershell.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"",
             "fi"
-        ) -join "`n" | Out-File $shim -Encoding ASCII
+        ) -join "`n" | Out-File $shim -Encoding ASCII -NoNewline
```

Comment:
Why this?

Reply:
No. Newlines in the middle of strings are not affected.
If this parameter is not added, `\r\n` will be automatically added to the end of the shell file. Causes an error to run in WSL.

Reply:
As shown. `0D 0A` is `\r\n`

![image](https://user-images.githubusercontent.com/4396864/148684027-4614cc7f-590f-4935-a075-43f9cc71dbae.png)

Reply:
In that case it will be needed for all sh shims (there are 5 in total).

Reply:
I want to test other sh shims.
How can I regenerate all shims?

Reply:
Other shims are created only in certain conditions, i.e. when a bin field has batch, java or python files. For the purpose of test, you can manually create the shims by running the relevant PowerShell commands.

Reply:
I modified other sh shims.
Thanks for your help.